src/api/kube.ts

```diff
@@ -568,24 +568,20 @@ export class KubeHelper {
     }
 
     if (!res || !res.body || !res.body.items) {
-      throw new Error(`Get pods by selector "${selector}" returned an invalid response`)
+      throw new Error(`Get pods by selector "${selector}" returned an invalid response.`)
     }
 
+    // No pods found by the specified selector. So, it's not ready.
     if (res.body.items.length < 1) {
-      // No pods found by the specified selector. So, it's not ready.
       return 'False'
     }
 
     if (res.body.items.length > 1) {
-      throw new Error(`Get pods by selector "${selector}" returned ${res.body.items.length} pods (1 was expected)`)
-    }
-
-    if (!res.body.items[0].status) {
-      throw new Error(`Get pods by selector "${selector}" returned a pod with an invalid state`)
+      throw new Error(`Get pods by selector "${selector}" returned ${res.body.items.length} pods (1 was expected).`)
     }
 
-    if (!res.body.items[0].status.conditions || !(res.body.items[0].status.conditions.length > 0)) {
-      throw new Error(`Get pods by selector "${selector}" returned a pod with an invalid status.conditions`)
+    if (!res.body.items[0].status || !res.body.items[0].status.conditions || !(res.body.items[0].status.conditions.length > 0)) {
+      return
```

Comment:
please return `undefined` explicitly

Reply:
I can't otherwise we will have lint errors

Reply:
:disappointed: